microsoft-edge/webview2/concepts/platform-components.md

```diff
@@ -0,0 +1,322 @@
+---
+title: Overview of the components of the WebView2 platform
+description: Which parts of WebView2 reside on the Dev and user machine.  How native code and controls interact with web code and the WebView2 control.
+author: MSEdgeTeam
+ms.author: msedgedevrel
+ms.topic: conceptual
+ms.service: microsoft-edge
+ms.subservice: webview
+ms.date: 04/21/2023
+---
+# Overview of the components of the WebView2 platform
+
+To add WebView2 to your app, you use the WebView2 SDK on your development machine, and distribute the WebView2 Runtime to user machines.  The following diagram shows the high-level WebView2 components on your development machine and user machines.
+
+![Full diagram of WebView2](../index-images/full-diagram.png)
+<!-- .png is owned by index.md, also linked to by platform-components.md -->
+
+Developing a WebView2 app involves software residing in the following places:
+
+| Location | Description |
+|---|---|
+| Dev machine | You use a Visual Studio project that includes the WebView2 SDK.  The SDK includes the WebView2 Runtime, which is an embedded web browser component used for the WebView2 control instances in your host app. |
+| Distributing the app and Runtime | There are several ways to deliver the always up-to-date Evergreen version of the WebView2 Runtime to user machines, with several levels of Internet connectivity supported.  Some scenarios benefit from distributing a specific, fixed-version WebView2 Runtime. |
+| User machine | Your host app on user machines includes instances of the WebView2 control, which uses the WebView2 Runtime. |
+| Resources | The present documentation; the WebView2Samples repo including basic Getting Started WebView2 projects and more full-featured Sample projects; the WebView2Announcements repo; and the WebView2Feedback repo. |
+
+
+<!-- todo: like the above table, talk through each box of the full diagram: -->
+
+Details of the Dev machine:
+*  Visual Studio project -- Use a Visual Studio project template to create a standard platform app, and then add the WebView2 SDK to the project as a NuGet package.
+   *  Layout designer -- lay out your controls in Visual Studio.
+      *  WebView2 control instances -- web content areas of your app.  The app's web-side code runs in this control.
+      *  Native control instances -- native controls and panes of your app.
+   *  WebView2 SDK
+      *  Per-platform WebView2 APIs <!-- todo: add to diagram --> including CoreWebView2, CoreWebView2Controller, CoreWebView2Environment.  Primarily called by native-side code.
+      *  `AddHostObjectToScript` -- enables exposing platform APIs and WebView2 APIs to JavaScript code.
+      *  [JavaScript APIs](/microsoft-edge/webview2/webview2-api-reference#javascript) (WebView2Script package) -- called by web-side code to communicate with the host application.
+   *  Platform APIs -- non-WebView2 APIs provided by the platform; can be exposed to web-side code. <!-- todo: add to diagram -->
```

Comment:
add to diagram: Platform APIs -- non-WebView2 APIs provided by the platform; can be exposed to web-side code.